storage/innobase/xtrabackup/src/space_map.cc

```diff
@@ -65,22 +65,30 @@ Tablespace_map &Tablespace_map::instance() { return (static_tablespace_map); }
 /** Scan I_S.FILES for extended tablespaces.
 @param[in]  connection MySQL connection object */
 void Tablespace_map::scan(MYSQL *connection) {
+  /*combine General and single tablespace*/
   const char *query =
-      "SELECT FILE_NAME, TABLESPACE_NAME, FILE_TYPE"
-      "  FROM INFORMATION_SCHEMA.FILES"
-      "  WHERE ENGINE = 'InnoDB'"
-      "    AND STATUS = 'NORMAL'"
-      "    AND FILE_TYPE <> 'TEMPORARY'"
-      "    AND FILE_ID <> 0";
+      "SELECT T2.PATH, "
+      "       T2.NAME, "
+      "       T1.SPACE_TYPE "
+      "FROM   INFORMATION_SCHEMA.INNODB_TABLESPACES T1 "
+      "       JOIN INFORMATION_SCHEMA.INNODB_TABLESPACES_BRIEF T2 USING "
+      "(SPACE) "
+      "WHERE  T1.SPACE_TYPE = 'Single' "
+      "UNION "
+      "SELECT T2.PATH, "
+      "       SUBSTRING_INDEX(SUBSTRING_INDEX(T2.PATH, '/', -1), '.', 1) NAME, "
```

Comment:
How does the general tablespace query part work for general tablespaces with datadir? Can you paste output here?

Reply:
This query has two GT created with and without data directory.

mysql> select T1.NAME,T2.PATH,SUBSTRING_INDEX(SUBSTRING_INDEX(T2.PATH, '/', -1),'.',1) NAME,T1.SPACE_TYPE from INNODB_TABLESPACES T1 join INNODB_TABLESPACES_BRIEF T2  using (space) where T1.SPACE_TYPE='General';
+------+-------------------------------+------+------------+
| NAME | PATH                          | NAME | SPACE_TYPE |
+------+-------------------------------+------+------------+
| ts2  | /Users/rahulmalik/tmp/ts1.ibd | ts1  | General    |
| ts3  | ts3.ibd                       | ts3  | General    |
+------+-------------------------------+------+------------+
2 rows in set (0.01 sec)

mysql>

Reply:
How does copy-back skip the local general tablespaces?

Reply:
It tablespaces are in data_directory they are not added in the map.
So JSON file will not have entry.